[DOCS] Puts regression model info into a separated section

szabosteve opened the pull request:
This PR adds a section title to the paragraph about regression models in the regression conceptual documentation. This makes it easier to link directly to the section about the used model.

Preview: http://stack-docs_723.docs-preview.app.elstc.co/guide/en/elastic-stack-overview/master/dfa-regression.html

lcawl commented on docs/en/stack/ml/df-analytics/dfa-regression.asciidoc:
```diff
@@ -71,14 +71,20 @@ the {depvar} by combining the predictions from multiple base learners –
+
+[discrete]
+[[dfa-regression-model]]
+===== {regression-cap} models
```
Is model the right term here?  In other places, we talk about creating models and model snapshots, so that's what I think of when I read the term. Whereas it seems to me this section is more about the algorithms that we use to create model.

```suggestion
===== {regression-cap} algorithms
```

lcawl commented on docs/en/stack/ml/df-analytics/dfa-regression.asciidoc:
```diff
@@ -71,14 +71,20 @@ the {depvar} by combining the predictions from multiple base learners –
+called extreme gradient boost (XGboost) which combines decision trees with 
+gradient boosting methodologies.
 
 {regression-cap} works as a batch analysis. If new data comes into your index, 
```
> Regression works as a batch analysis...

I think this paragraph doesn't pertain to this section and belongs in the previous one.

szabosteve commented:
@lcawl I addressed your feedback and opened another PR that changes the title of the same section in the classification docs. This way the structure of the pieces will be in line.
https://github.com/elastic/stack-docs/pull/727
